crates/ruma-common/src/events/room/message/reply.rs

```diff
@@ -97,17 +95,20 @@ fn formatted_or_plain_body(
 /// [HTML tags and attributes]: https://spec.matrix.org/v1.4/client-server-api/#mroommessage-msgtypes
 /// [rich reply fallbacks]: https://spec.matrix.org/v1.4/client-server-api/#fallbacks-for-rich-replies
 pub fn plain_and_formatted_reply_body(
-    body: impl fmt::Display,
-    formatted: Option<impl fmt::Display>,
+    formatted: Option<&FormattedBody>,
+    body: &str,
```

Comment:
Then you only need to change it like this:

```suggestion
    body: &str,
    formatted: Option<impl fmt::Display>,
```